corehq/warehouse/models/meta.py

```diff
@@ -0,0 +1,61 @@
+from django.db import models
+
+from django_fsm import FSMField, transition
+
+from ..states import (
+    FACT_TABLE_NEEDS_UPDATING,
+    FACT_TABLE_BATCH_DUMPED,
+    FACT_TABLE_READY,
+    FACT_TABLE_FAILED,
+    FACT_TABLE_DECOMMISSIONED,
+)
+
+
+class FactTableState(models.Model):
```

Comment:
i don't remember if we discussed this or not but if we end up taking the batched approach for dims we should probably just have one generic table that stores the state of each batch for each dim/fact